### PR TITLE
Futures checks

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/exchanges/exchange_builder.py [hunk 1]
```diff
@@ -120,6 +122,18 @@ async def _build_trading_modes(self, trading_mode_class):
             self.logger.error(f"An error occurred when initializing trading mode : {e}")
             raise e
 
+    def _ensure_exchange_compatibility(self):
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_builder.py:125]: 👍

File: octobot_trading/exchanges/exchange_factory.py [hunk 2]
```diff
@@ -76,6 +77,18 @@ async def initialize_real_exchange(exchange_manager):
             await exchanges.search_and_create_websocket(exchange_manager)
 
 
+def _ensure_exchange_validity(exchange_manager):
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_factory.py:80]: 👍

File: octobot_trading/exchanges/util/exchange_util.py [hunk 3]
```diff
@@ -135,28 +135,42 @@ def _get_minimal_exchange_config(exchange_name, exchange_config):
...
     local_exchange_manager = exchange_manager.ExchangeManager(
         _get_minimal_exchange_config(exchange_name, exchange_config),
         exchange_name
     )
     local_exchange_manager.tentacles_setup_config = tentacles_setup_config
     # Avoid exchange authentication error log
-    local_exchange_manager.is_collecting = True
+    local_exchange_manager.check_credentials = False
+    local_exchange_manager.is_sandboxed = is_sandboxed
     await exchange_factory.search_and_create_spot_exchange(local_exchange_manager)
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_util.py:154]: spot exchange? Is it working when using future exchange?

File: tests/exchanges/util/test_exchange_util.py [hunk 4]
```diff
@@ -43,14 +43,28 @@ def exchange_config():
...
+        assert auth is True
         assert error is None
         is_valid_account_mock.assert_called_once()
     with mock.patch.object(trading_backend.exchanges.Binance, "is_valid_account",
                            mock.AsyncMock(return_value=(False, "plop"))) as is_valid_account_mock:
-        compatible, error = await exchanges.is_compatible_account("binance", exchange_config, tentacles_setup_config)
+        compatible, auth, error = await exchanges.is_compatible_account("binance", exchange_config,
+                                                                        tentacles_setup_config, False)
+        # still True as on spot trading
+        assert compatible is True
```
Comment (Herklos) [inline tests/exchanges/util/test_exchange_util.py:57]: 👍

File: octobot_trading/exchanges/util/exchange_util.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/util/exchange_util.py]: Right, for now it works as is but we should not keep it, I changed it

File: octobot_trading/exchanges/exchange_builder.py [hunk 5]
```diff
@@ -52,19 +49,17 @@ async def build(self):
 
     async def _build_exchange_manager(self):
         trading_mode_class = None
-        self.exchange_manager.tentacles_setup_config = self._tentacles_setup_config
-        self.exchange_manager.bot_id = self._bot_id
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_builder.py:56]: 👍

File: octobot_trading/exchanges/exchange_builder.py [hunk 6]
```diff
@@ -165,6 +160,16 @@ def is_real(self):
         self.exchange_manager.trader = exchanges.Trader(self.config, self.exchange_manager)
         return self
 
+    def is_using_exchange_type(self, exchange_type):
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_builder.py:163]: 👍

File: octobot_trading/exchanges/util/exchange_util.py [hunk 7]
```diff
@@ -142,24 +144,19 @@ async def is_compatible_account(exchange_name: str, exchange_config: dict, tenta
...
-        local_exchange_manager.exchange.connector.set_sandbox_mode(local_exchange_manager.is_sandboxed)
+    local_exchange_manager = await builder.use_tentacles_setup_config(tentacles_setup_config) \
+        .is_checking_credentials(False) \
+        .is_sandboxed(is_sandboxed) \
+        .is_using_exchange_type(exchange_type) \
+        .is_exchange_only() \
+        .is_rest_only() \
+        .is_loading_markets(False) \
+        .disable_trading_mode() \
+        .build()
```
Comment (Herklos) [inline octobot_trading/exchanges/util/exchange_util.py:159]: 👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**